### ISSUE TITLE
Unloading of Model from memory in Python API

### ISSUE THREAD

**Issue by gsarathkumar92** [openvinotoolkit/openvino#11978]
Users can work on as many networks as they need and use them simultaneously (up to the limitation of the hardware resources). Unload and load the model whenever it is necessary based on application.

**Comment by gsarathkumar92** [comment]
Similar to this feature - https://github.com/openvinotoolkit/openvino/pull/8353 in Python API

**Comment by brmarkus** [comment]
@gsarathkumar92  can you maybe rephrase your question, please?
You want to explicitly unload a model from memory after loading or importing it, when finished using the models?

**Comment by ilya-lavrenov** [comment]
Hi @gsarathkumar92 
I think you can use Model cache feature https://docs.openvino.ai/latest/openvino_docs_OV_UG_Model_caching_overview.html. Once you don't need any model, you can unload it. Once you need it once again - just reload from disk, but since the compiled model is already cached, the loading / compilation will be much faster.

**Comment by brmarkus** [comment]
> Once you don't need any model, you can unload it

How? How to unload the model? Which API(s) to call to unload the model and free-up memory?

**Comment by ilya-lavrenov** [comment]
> > Once you don't need any model, you can unload it
> 
> How? How to unload the model? Which API(s) to call to unload the model and free-up memory?

```c++
ov::CompiledModel model = ...;
model = {};
```
Just destroy the object.

**Comment by jgespino** [comment]
Closing, please re-open if additional assistance is needed.

**Comment by gsarathkumar92** [comment]
@ilya-lavrenov  If we just destroy the object will that be unloaded from the GPU?
